### ISSUE TITLE
Proper expression language (EL) for conditions 

### ISSUE THREAD

**Issue by alexsnaps** [Kuadrant/limitador#102]
Currently `conditions` for a `limit` have the current format:

```yaml
  conditions:
    - "req.method == GET"
```

Where left-hand side (lhs) _has to be_ the `variable` to look up and the right-hand side (rhs) the value it'll be tested against. Currently we _only_ support the equality operator  (`==`) and further more, _only_ the format where it has a white space before _and_ after the operator is [actually supported](https://github.com/Kuadrant/limitador/blob/main/limitador/src/limit.rs#L124-L131) (where more spaces will probably _not_ have [the result one would expect](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1a874510d22ec6c0d27294e5d303d56f)).

I started some work on parsing these condition using a proper scanner which turns the input `&str` (each condition) into tokens (identifier, operator & ... a `String` literal). But right now it still has to fallback to position to know which is the identifier (lhs) and which is the value (rhs). I think that's less optimal and possibly creates really surprising things to read, as e.g. `get == GET` would be a valid condition.

I'm proposing to have `String` literals just be… string literals as one would know them from different programing languages or... yaml for this matter (i.e. surrounded by quotation (`"` or `'`) marks). That would achieve a few things: 

 - White spaces don't matter anymore (or the lack thereof, see #94)
 - Positioning doesn't matter, `method == "GET"` and `"GET" == method` are equivalent, as one would expect as `==` is commutative.
 - Reading the condition is straightforward and less error prone

What would that EL look like tho? 
 - Should we use double (`"`) or/and single (`'`) quotes? 
 - Based on [my reading](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-msg-config-route-v3-ratelimit-action-generickey), there can only ever be a `String` value in there, so we don't need to support `Number` literals, right? 
 - Use backslah (`\`) for escaping? Using both single & double quotes?

The downside is that would be a breaking change. Existing `limits.yaml` file would be "wrong". I already wanted to eagerly parse the conditions when parsing the `limits.yaml` file (unlike the lazy way of today). So we could give a good error message to users when the scanner would have a `[TokenType::Identifier, TokenType::Operator, TokenType::Identifier]` when it'd expect one (and probably the last one) to be a `TokenType::Literal` instead. 

## tl;dr

Have condition be in the format of:

```yaml
  conditions:
    - "req.method == 'GET'"
```

### todo

 - [ ] Validate conditions _when parsing the limit file_ see #100
 - [ ] Back `TryFrom<&str> for Condition` by a proper `Scanner` impl.

**Comment by guicassolato** [comment]
Authorino's [`when` conditions](https://github.com/Kuadrant/authorino/blob/main/docs/features.md#common-feature-conditions-when) are represented as structured expressions always composed of 3 separate fields: `selector`, `operator` and `value`. One good thing of using structured fields is that, to an extend, they can be validated at the level of the YAML spec (OAS, in the case of a CRD), rather then only parsed/compiled by the component at runtime.

Authorino's selectors are JSON paths that not only represent a value (fetched from the [Authorization JSON](https://github.com/Kuadrant/authorino/blob/main/docs/architecture.md#the-authorization-json)) but also accept functions (“[string modifiers](https://github.com/Kuadrant/authorino/blob/main/docs/features.md#string-modifiers)”).

**Comment by alexsnaps** [comment]
Looking at where this comes from today, _and if we wanted to move away from the conditions requiring an evaluation, the "natural" way of expressing would be like: 

```yaml
{
  "descriptor_key": "method",
  "exact_match": "POST"
}
```

**Comment by alexsnaps** [comment]
Not quite sure what people's preference is here, but I'll come with mine… 
Since Limitador doesn't dependent on any way on the k8s in any way, I'd stick to what is there:
 - plain strings that are being evaluated: `var == "value"`
 - support string types _for now_ (as this is really what's going on underneath): i.e. it'd be `GET == "1"`
 - support the "old" (i.e. current) behavior for some time, we need to come up with a deprecation strategy, and probably link to some content on the website on how to migrate your current config
 - add support for `!=` see #18 

There are a few reason for this, but mostly this is the most straight forward change, but mostly addresses the type system clearly… unlike splitting the condition in 3 distinct parts… how is the type of the value expressed? And I know there is only `string` for now, but I don't want to close any doors here at this stage. Finally we can always build more "k8s native" expressive things at the CRD levels…
